trieste/bayesian_optimizer.py

```diff
@@ -788,152 +822,152 @@ def _write_summary_init(
-            logging.histogram("query_points/euclidean_distances", lambda: pdist(query_points))
-
-        if pd and sns and logging.include_summary("query_points/_pairplot"):
-            columns = [f"x{i}" for i in range(tf.shape(query_points)[1])]
-            qp_preds = query_points
-            for tag in datasets:
-                qp_preds = tf.concat(
-                    [qp_preds, tf.cast(models[tag].predict(query_points)[0], query_points.dtype)], 1
+            if tf.size(datasets[tag].observations) > 0:
+                logging.scalar(
+                    f"{tag}.observation{suffix}/best_overall",
+                    np.min(datasets[tag].observations[..., i]),
                 )
-                columns.append(f"{tag} predicted")
-            query_new_df = pd.DataFrame(qp_preds, columns=columns).applymap(float)
-            query_new_df["query points"] = "new"
-            query_plot_df = pd.concat(
-                (query_plot_dfs.get(0), query_new_df), copy=False, ignore_index=True
+
+        if logging.include_summary(f"{tag}.observations/_pairplot") and (
+            pd and sns and output_dim >= 2
+        ):
+            columns = [f"x{i}" for i in range(output_dim)]
+            observation_new_df = pd.DataFrame(
+                tagged_output[tag].observations, columns=columns
+            ).applymap(float)
+            observation_new_df["observations"] = "new"
+            observation_plot_df = pd.concat(
+                (observation_plot_dfs.get(tag), observation_new_df),
+                copy=False,
+                ignore_index=True,
             )
-            pairplot = sns.pairplot(query_plot_df, hue="query points", height=2.25)
-            padding = 0.025 * (self._search_space.upper - self._search_space.lower)
-            upper_limits = self._search_space.upper + padding
-            lower_limits = self._search_space.lower - padding
-            for i in range(self._search_space.dimension):
-                pairplot.axes[0, i].set_xlim((lower_limits[i], upper_limits[i]))
-                pairplot.axes[i, 0].set_ylim((lower_limits[i], upper_limits[i]))
-            logging.pyplot("query_points/_pairplot", pairplot.fig)
-            query_plot_df["query points"] = "old"
-            query_plot_dfs[0] = query_plot_df
-
-        logging.scalar("wallclock/step", total_step_wallclock_timer.time)
-        logging.scalar(
-            "wallclock/query_point_generation",
-            query_point_generation_timer.time,
-        )
-        logging.scalar(
-            "wallclock/model_fitting",
-            model_fitting_timer.time,
+            observation_plot_df["pareto"] = non_dominated(datasets[tag].observations)[1] == 0
+            observation_plot_df["observation type"] = observation_plot_df.apply(
+                lambda x: x["observations"] + x["pareto"] * " (non-dominated)",
+                axis=1,
+            )
+            pairplot = sns.pairplot(
```

Comment:
a bit trickier is that if you have constraints, then your Pareto front is different, see 

https://secondmind-labs.github.io/trieste/0.12.0/notebooks/multi_objective_ehvi.html#Acquisition-function-for-multiple-objectives-and-constraints

Reply:
Not immediately obvious how we can do this out of the box: all the default logger knows is that there are multiple tagged datasets (one of which is probably, but not necessarily, called OBJECTIVE), not the relation between them. Any ideas?

Suggest creating an issue for this and fixing it in a separate PR.

Reply:
(We could delegate the plotting to the acquisition function builder, but that will require some more infrastructure, so definitely makes sense to do in a separate PR.)

Reply:
that's fair - I thought the EHVI etc. were expected specifically an OBJECTIVE tag but you can specify whatever tag you want as `objective_tag`.

Reply:
I think this makes sense. Showing pairs plot of predictions with markers based on multi-objective considerations only makes sense if it matches your acquisition function builder.